sdk/src/settings/mod.rs

```diff
@@ -160,67 +176,126 @@ impl SettingsValidate for Trust {
         }
     }
 }
 
 impl SettingsValidate for Core {
     fn validate(&self) -> Result<()> {
-        match self.hash_alg.as_str() {
-            "sha256" | "sha384" | "sha512" => Ok(()),
-            _ => Err(Error::UnsupportedType),
-        }
+        Ok(())
     }
 }
 
-// Settings for verification options
+/// Settings to configure the verification process.
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-#[allow(unused)]
-pub(crate) struct Verify {
-    verify_after_reading: bool,
-    verify_after_sign: bool,
-    verify_trust: bool,
-    verify_timestamp_trust: bool,
-    ocsp_fetch: bool,
-    remote_manifest_fetch: bool,
-    check_ingredient_trust: bool,
-    skip_ingredient_conflict_resolution: bool,
-    strict_v1_validation: bool,
+pub struct Verify {
+    /// Whether to verify the manifest after reading in the [`Reader`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Disabling validation can improve reading performance, BUT it carries the risk of reading an invalid
+    /// manifest.
+    /// </div>
+    ///
+    /// [`Reader`]: crate::Reader
+    pub verify_after_reading: bool,
+    /// Whether to verify the manifest after signing in the [`Builder`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Disabling validation can improve signing performance, BUT it carries the risk of signing an invalid
+    /// manifest.
+    /// </div>
+    ///
+    /// [`Builder`]: crate::Builder
+    pub verify_after_sign: bool,
+    /// Whether to verify certificates against the trust lists specified in [`Trust`]. To configure
+    /// timestamp certificate verification, see [`Verify::verify_timestamp_trust`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Verifying trust is REQUIRED by the C2PA spec. This option should only be used for development or testing.
+    /// </div>
+    pub verify_trust: bool,
+    /// Whether to verify the timestamp certificates against the trust lists specified in [`Trust`].
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// Verifying timestamp trust is REQUIRED by the C2PA spec. This option should only be used for development or testing.
+    /// </div>
+    pub verify_timestamp_trust: bool,
+    /// Whether to fetch the certificates OCSP status during validation. This setting will fetch OCSP for all certificates
+    /// even if they are stapled with a [`CertificateStatus`] assertion.
+    ///
+    /// The default value is false.
+    ///
+    /// [`CertificateStatus`]: crate::assertions::CertificateStatus
+    pub ocsp_fetch: bool,
+    /// Whether to fetch remote manifests during reading in the [`Reader`].
+    ///
+    /// The default value is true.
+    ///
+    /// [`Reader`]: crate::Reader
+    pub remote_manifest_fetch: bool,
```

Comment:
Maybe we should move this and `verify.verify_after_reading` to a `ReaderSettings`? I think there's a few other settings we can move to `BuilderSettings` and `ReaderSettings` from here. Any thoughts?

Reply:
`remote_manifest_fetch` feels right for me in ReaderSettings. Because it configures the behavior of the Reader, aka if it retrieves remote manifests or not.

Reply:
It may be a good idea to separate ingredient settings into its own thing. This would be especially useful for configuring thumbnail generation. Maybe it should be a subset of `ReadSettings`?

Reply:
There can definitely be different options people want for ingredient vs claim thumbnails.

Reply:
Hm, there's a lot of overlap with the `Reader`/`Builder`/`Ingredient` settings, such as for `remote_manifest_fetch`. We also need to come up with a strategy to preserve backwards compatibility if we were to move these fields. I'm thinking we may want to leave that up to a separate PR.